tests/test_emitter.py

```diff
@@ -74,38 +73,76 @@ def setup_teardown(tmpdir):
 
 
 def start_watching(path=None, use_full_emitter=False, recursive=True):
-    path = p('') if path is None else path
+    # todo: check if other platforms expect the trailing slash (e.g. `p('')`)
+    path = p() if path is None else path
     global emitter
     if platform.is_linux() and use_full_emitter:
         emitter = InotifyFullEmitter(event_queue, ObservedWatch(path, recursive=recursive))
     else:
         emitter = Emitter(event_queue, ObservedWatch(path, recursive=recursive))
 
-    if platform.is_darwin():
-        # FSEvents will report old events (like create for mkdtemp in test
-        # setup. Waiting for a considerable time seems to 'flush' the events.
-        time.sleep(10)
     emitter.start()
 
+    if platform.is_darwin():
+        # FSEvents _may_ report the event for the creation of `tmpdir`,
+        # however, we're racing with fseventd there - if other filesystem
+        # events happened _after_ `tmpdir` was created, but _before_ we
+        # created the emitter then we won't get this event.
+        # As such, let's create a sentinel event that tells us that we are
+        # good to go.
+        sentinel_file = os.path.join(path, '.sentinel')
```

Comment:
This will fail if `path` is bytes, for instance in `test_passing_bytes_should_give_bytes`.